NETDefault/UnitTestingWithNUnit&Moq/UserServiceTests.cs

```diff
@@ -76,19 +67,31 @@ public void GetDashboard_WhenHobbiesFound_ItShouldBeReturned(string[] hobbies, s
         public void GetDashboard_WhenNotificationsFound_ItShouldBeReturned(int? count, string expectedNotificationString)
         {
             // Arrange
-            SetupGetUserIdByUserName(_knownUsername);
-            SetupGetHobbiesByUserId(new string[0] { });
+            SetupGetUserIdByUserName();
             SetupGetNotificationsByUserId(count);
 
             // Act
-            var result = (string)_subject.GetDashboardByUserName(_knownUsername);
+            var result = _subject.GetDashboardByUserName(KnownUsername);
 
             // Assert
             Assert.That(result.Contains(expectedNotificationString));
+        }
+        
+        [Test]
+        public void GetDashboard_WhenExceptionIsThrown_ItShouldBeLogged()
```

Comment:
I think this logging assertion can also be added to the `GetDashboard_WhenCustomerNameNotFound_ItShouldThrow` test. I don't think you need two separate tests to test the same scenario.

Reply:
It tests two different behaviors though, logging is separate from throwing, so also tested differently.

Normally the setup would also be different, but as I did the minimal updates for now it has the same arrange steps (which isn't necessarily bad in tests either).